src/main/java/com/redhat/devtools/intellij/knative/tree/KnFunctionsTreeStructure.java

```diff
@@ -230,7 +231,8 @@ private VirtualFile[] getModuleRoots(Module module) {
             roots.add(getModuleRootAsDirectory(contentRoot));
         }
         if (roots.isEmpty()) {
-            roots.add(getModuleRootAsDirectory(LocalFileSystem.getInstance().findFileByPath(new File(module.getModuleFilePath()).getParent())));
+            VirtualFile moduleDir = ProjectUtil.guessModuleDir(module);
+            roots.add(getModuleRootAsDirectory(moduleDir));
```

Comment:
@sbouchet, @jeffmaury, @lstocchi: I am very unsure about this replacement. Can you please enlighten me what this tried to do?